tests/ssl_test.py

```diff
@@ -371,3 +372,24 @@ def client(addr):
         peer, _ = server_tls.accept()
         assert peer.recv(64) == expected
         peer.close()
+
+    @tests.skip_if(sys.version_info[:2] < (3, 7))
+    def test_client_check_hostname(self):
+        # stdlib API compatibility
+        # https://github.com/eventlet/eventlet/issues/567
+        def serve(listener):
+            sock, addr = listener.accept()
+            sock.recv(64)
+            sock.sendall(b"response")
+            sock.close()
+
+        listener = listen_ssl_socket()
+        server_coro = eventlet.spawn(serve, listener)
+        client = ssl.wrap_socket(
+            eventlet.connect(listener.getsockname()),
+            check_hostname=True,
```

Comment:
This doesn't seem to be how CPython's `ssl.wrap_socket()` works:
``` python
>>> import socket, ssl
>>> sock = socket.socket()
>>> sock.connect(('github.com', 443))
>>> ssock = ssl.wrap_socket(sock, check_hostname=True, verify_mode=ssl.CERT_REQUIRED)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: wrap_socket() got an unexpected keyword argument 'check_hostname'
```
WDYT about something like
``` diff
diff --git a/tests/ssl_test.py b/tests/ssl_test.py
index c5b5b49..dda82ff 100644
--- a/tests/ssl_test.py
+++ b/tests/ssl_test.py
@@ -385,11 +385,14 @@ class SSLTest(tests.LimitedTestCase):
 
         listener = listen_ssl_socket()
         server_coro = eventlet.spawn(serve, listener)
-        client = ssl.wrap_socket(
-            eventlet.connect(listener.getsockname()),
-            check_hostname=True,
-            verify_mode=ssl.CERT_REQUIRED,
-        )
+        ctx = ssl.SSLContext()
+        ctx.verify_mode = ssl.CERT_REQUIRED
+        ctx.check_hostname = True
+        ctx.load_verify_locations(tests.certificate_file)
+        client = ctx.wrap_socket(
+            socket.socket(socket.AF_INET, socket.SOCK_STREAM),
+            server_hostname="Test")
+        client.connect(listener.getsockname())
         client.send(b"check_hostname works")
         client.recv(64)
         server_coro.wait()

```
?